test/compression.js

```diff
@@ -676,6 +676,35 @@ describe('compression()', function () {
       .end()
     })
   })
+
+  describe('when callbacks are used', function () {
+    it('should call the passed callbacks', function(done){
+      var callbacks = 0;
+      var server = createServer({ threshold: '1kb' }, function (req, res) {
+        res.setHeader('Content-Type', 'text/plain')
+        res.write('Hello', null, function(){
+          callbacks++
+          res.flush()
```

Comment:
Gotcha. So perhaps that would call for two different tests? Testing the callback when not flushed vs testing with flushed? DO you think that would be a meaningful test?